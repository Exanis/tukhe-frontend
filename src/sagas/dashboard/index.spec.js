import watcherSaga, * as sagas from './index';

it('Should watch every sagas', () => {
    const gen = watcherSaga();

    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should process saga as expected', () => {
    const gen = sagas.startProcessSaga({
        payload: {
            target: jest.fn(),
            interval: 'test interval',
            widget: 'test widget'
        }
    });
    const forkExecutor = gen.next();

    expect(forkExecutor).toMatchSnapshot();
    const forkFunction = forkExecutor.value.payload.fn();

    expect(forkFunction.next()).toMatchSnapshot();
    expect(forkFunction.next()).toMatchSnapshot();
    forkFunction.throw('end');
    expect(forkFunction.next().done).toBeTruthy();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next({
        payload: {
            widget: 'test widget'
        }
    })).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run select dashboard saga as expected', () => {
    const gen = sagas.selectDashboardSaga({
        payload: 'test'
    });
    const request = gen.next();

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({
        data: []
    })).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run select dashboard saga as expected when there is an error', () => {
    const gen = sagas.selectDashboardSaga({
        payload: 'test'
    });
    const request = gen.next();

    expect(request).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run dashboard load saga as expected', () => {
    const gen = sagas.loadDashboardSaga();
    const fetch = gen.next();

    expect(fetch).toMatchSnapshot();
    expect(fetch.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({data: [
        {
            uuid: 'test'
        }
    ]})).toMatchSnapshot();
    expect(gen.next({
        currentDashboard: null
    })).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run dashboard load saga as expected when there is a dashboard', () => {
    const gen = sagas.loadDashboardSaga();
    const fetch = gen.next();

    expect(fetch).toMatchSnapshot();
    expect(fetch.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({data: [
        {
            uuid: 'test'
        }
    ]})).toMatchSnapshot();
    expect(gen.next({
        currentDashboard: 'something'
    })).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run dashboard load saga as expected when there is an error', () => {
    const gen = sagas.loadDashboardSaga();
    const fetch = gen.next();

    expect(fetch).toMatchSnapshot();
    gen.throw('test');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run create dashboard saga as expected', () => {
    const gen = sagas.createDashboardSaga({
        payload: 'home'
    });
    const request = gen.next();

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run create saga as expected when there is an error', () => {
    const gen = sagas.createDashboardSaga({
        payload: 'home'
    });

    gen.next();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run create widget saga as expected', () => {
    const gen = sagas.createWidgetSaga({
        payload: {
            title: 'test',
            header: true,
            type: 'twitter_feed',
            config: {
                target: 'test'
            },
            accounts: [
                'test',
                'test2'
            ]
        }
    });
    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'test'
    });

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();

    const associate = gen.next({data: {uuid: 'test'}});

    expect(associate).toMatchSnapshot();
    expect(associate.value.payload[0]).toMatchSnapshot();
    expect(associate.value.payload[1]).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run create widget saga as expected when there is an error', () => {
    const gen = sagas.createWidgetSaga({
        payload: {
            title: 'test',
            header: true,
            type: 'twitter_feed',
            config: {
                target: 'test'
            },
            accounts: [
                'test',
                'test2'
            ]
        }
    });
    expect(gen.next()).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run update layout saga as expected', () => {
    const gen = sagas.updateDashboardLayoutSaga({
        payload: {}
    });
    expect(gen.next()).toMatchSnapshot();
    const request = gen.next({
        currentDashboard: 'test'
    });

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run update layout saga as expected when there is an error', () => {
    const gen = sagas.updateDashboardLayoutSaga({
        payload: {}
    });
    expect(gen.next()).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run refresh with account saga as expected', () => {
    const gen = sagas.refreshWidgetWithAccountSaga({
        payload: 'test'
    });

    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'test'
    });

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({
        data: []
    })).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run refresh with account saga as expected when there is an error', () => {
    const gen = sagas.refreshWidgetWithAccountSaga({
        payload: 'test'
    });

    expect(gen.next()).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run refresh without account saga as expected', () => {
    const gen = sagas.refreshWidgetWithoutAccountSaga({
        payload: 'test'
    });

    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'test'
    });

    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next({
        data: []
    })).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run refresh without account saga as expected when there is an error', () => {
    const gen = sagas.refreshWidgetWithoutAccountSaga({
        payload: 'test'
    });

    expect(gen.next()).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run action with account as expected when param is not a formdata', () => {
    const gen = sagas.widgetActionWithAccountSaga({payload: {
        action: 'test',
        params: {}
    }});

    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'dashboard'
    });
    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run action with account as expected when param is a formdata', () => {
    const params = {
        action: 'test',
        params: new FormData(),
        callback: jest.fn()
    };
    const gen = sagas.widgetActionWithAccountSaga({payload: params});

    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'dashboard'
    });
    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run action with account as expected when param is not a formdata', () => {
    const gen = sagas.widgetActionWithAccountSaga({payload: {
        action: 'test',
        params: {}
    }});

    expect(gen.next()).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run action without account as expected when param is not a formdata', () => {
    const gen = sagas.widgetActionWithoutAccountSaga({payload: {
        action: 'test',
        params: {}
    }});

    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'dashboard'
    });
    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run action without account as expected when param is a formdata', () => {
    const params = {
        action: 'test',
        params: new FormData(),
        callback: jest.fn()
    };
    const gen = sagas.widgetActionWithoutAccountSaga({payload: params});

    expect(gen.next()).toMatchSnapshot();

    const request = gen.next({
        currentDashboard: 'dashboard'
    });
    expect(request).toMatchSnapshot();
    expect(request.value.payload.fn()).toMatchSnapshot();
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});

it('Should run action without account as expected when param is not a formdata', () => {
    const gen = sagas.widgetActionWithoutAccountSaga({payload: {
        action: 'test',
        params: {}
    }});

    expect(gen.next()).toMatchSnapshot();
    gen.throw('err');
    expect(gen.next()).toMatchSnapshot();
    expect(gen.next().done).toBeTruthy();
});
