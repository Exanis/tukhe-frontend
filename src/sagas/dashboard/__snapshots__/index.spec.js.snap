// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should process saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should process saga as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [MockFunction],
    },
    "type": "CALL",
  },
}
`;

exports[`Should process saga as expected 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        "test interval",
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should process saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "pattern": [Function],
    },
    "type": "TAKE",
  },
}
`;

exports[`Should process saga as expected 5`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": "@@redux-saga/SELF_CANCELLATION",
    "type": "CANCEL",
  },
}
`;

exports[`Should process saga as expected 6`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run action with account as expected when param is a formdata 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run action with account as expected when param is a formdata 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run action with account as expected when param is a formdata 3`] = `Promise {}`;

exports[`Should run action with account as expected when param is a formdata 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": null,
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run action with account as expected when param is not a formdata 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run action with account as expected when param is not a formdata 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run action with account as expected when param is not a formdata 3`] = `Promise {}`;

exports[`Should run action with account as expected when param is not a formdata 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run action with account as expected when param is not a formdata 5`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run action with account as expected when param is not a formdata 6`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run action without account as expected when param is a formdata 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run action without account as expected when param is a formdata 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run action without account as expected when param is a formdata 3`] = `Promise {}`;

exports[`Should run action without account as expected when param is a formdata 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": null,
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run action without account as expected when param is not a formdata 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run action without account as expected when param is not a formdata 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run action without account as expected when param is not a formdata 3`] = `Promise {}`;

exports[`Should run action without account as expected when param is not a formdata 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run action without account as expected when param is not a formdata 5`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run action without account as expected when param is not a formdata 6`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run create dashboard saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run create dashboard saga as expected 2`] = `Promise {}`;

exports[`Should run create dashboard saga as expected 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": undefined,
        "type": "dashboard/load/list",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run create dashboard saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": null,
        "type": "user/setToken",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run create saga as expected when there is an error 1`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run create widget saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run create widget saga as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run create widget saga as expected 3`] = `Promise {}`;

exports[`Should run create widget saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": true,
    "payload": Array [
      Promise {},
      Promise {},
    ],
    "type": "ALL",
  },
}
`;

exports[`Should run create widget saga as expected 5`] = `Promise {}`;

exports[`Should run create widget saga as expected 6`] = `Promise {}`;

exports[`Should run create widget saga as expected 7`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": "test",
        "type": "dashboard/select",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run create widget saga as expected when there is an error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run create widget saga as expected when there is an error 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run dashboard load saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run dashboard load saga as expected 2`] = `Promise {}`;

exports[`Should run dashboard load saga as expected 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run dashboard load saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": Array [
          Object {
            "uuid": "test",
          },
        ],
        "type": "dashboard/setDashboards",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run dashboard load saga as expected 5`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": "test",
        "type": "dashboard/select",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run dashboard load saga as expected when there is a dashboard 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run dashboard load saga as expected when there is a dashboard 2`] = `Promise {}`;

exports[`Should run dashboard load saga as expected when there is a dashboard 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run dashboard load saga as expected when there is a dashboard 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": Array [
          Object {
            "uuid": "test",
          },
        ],
        "type": "dashboard/setDashboards",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run dashboard load saga as expected when there is an error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run dashboard load saga as expected when there is an error 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run refresh with account saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run refresh with account saga as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run refresh with account saga as expected 3`] = `Promise {}`;

exports[`Should run refresh with account saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": Object {
          "data": Array [],
          "widget": "test",
        },
        "type": "dashboard/setData",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run refresh with account saga as expected when there is an error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run refresh with account saga as expected when there is an error 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run refresh without account saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run refresh without account saga as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run refresh without account saga as expected 3`] = `Promise {}`;

exports[`Should run refresh without account saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": Object {
          "data": Array [],
          "widget": "test",
        },
        "type": "dashboard/setData",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run refresh without account saga as expected when there is an error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run refresh without account saga as expected when there is an error 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run select dashboard saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run select dashboard saga as expected 2`] = `Promise {}`;

exports[`Should run select dashboard saga as expected 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": "test",
        "type": "dashboard/selectDashboard",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run select dashboard saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": Array [],
        "type": "dashboard/setWidgets",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run select dashboard saga as expected when there is an error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run select dashboard saga as expected when there is an error 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should run update layout saga as expected 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run update layout saga as expected 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
    "type": "CALL",
  },
}
`;

exports[`Should run update layout saga as expected 3`] = `Promise {}`;

exports[`Should run update layout saga as expected 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "action": Object {
        "payload": undefined,
        "type": "dashboard/load/list",
      },
      "channel": undefined,
    },
    "type": "PUT",
  },
}
`;

exports[`Should run update layout saga as expected when there is an error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [],
      "selector": [Function],
    },
    "type": "SELECT",
  },
}
`;

exports[`Should run update layout saga as expected when there is an error 2`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Should watch every sagas 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 5`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 6`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 7`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 8`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 9`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;

exports[`Should watch every sagas 10`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "combinator": false,
    "payload": Object {
      "args": Array [
        [Function],
        [Function],
      ],
      "context": null,
      "fn": [Function],
    },
    "type": "FORK",
  },
}
`;
